src/disk_interface.cc

```diff
@@ -266,22 +266,16 @@ FileReader::Status RealDiskInterface::ReadFile(const string& path,
 
 int RealDiskInterface::RemoveFile(const string& path) {
 #ifdef _WIN32
-  DWORD attributes = GetFileAttributes(path.c_str());
-  if (attributes == INVALID_FILE_ATTRIBUTES &&
-      GetLastError() == ERROR_FILE_NOT_FOUND) {
-    return 1;
-  }
-  if (attributes & FILE_ATTRIBUTE_READONLY) {
-    // On non-Windows systems remove will happily delete read-only files. On
-    // Windows Ninja should behave the same. See
-    // https://github.com/ninja-build/ninja/issues/1886
-    SetFileAttributes(path.c_str(), attributes & ~FILE_ATTRIBUTE_READONLY);
-  }
-  if (!DeleteFile(path.c_str())) {
-    Error("remove(%s): %s", path.c_str(), GetLastErrorString().c_str());
-    return -1;
```

Comment:
The `DeleteFile` code path is removed in favor of `remove()` by the last commit of this PR.  The idea is to minimize the net change from the code prior to #1892.  If we want to keep `DeleteFile` on Windows, we can just drop that commit.  I don't think `DeleteFile` is semantically much different than `remove()` on Windows.  It's not synchronous anyway.